frontend/src/components/party.js

```diff
@@ -0,0 +1,256 @@
+import "bootstrap/dist/css/bootstrap.css";
+import "../styles/modals.css";
+import Container from 'react-bootstrap/Container';
+import { Col, Row, Table } from "react-bootstrap";
+import {
+  DatatableWrapper,
+  TableBody,
+  TableColumnType,
+  TableHeader
+} from "react-bs-datatable";
+import axios from 'axios';
+import Button from 'react-bootstrap/Button';
+import React from "react";
+import Modal from 'react-bootstrap/Modal';
+import Form from 'react-bootstrap/Form';
+import toast, { Toaster } from 'react-hot-toast';
+import { useParams } from "react-router-dom";
+const someError = (msg) => toast.error("Oh No: " + msg);
+const itsOK = (msg) => toast.success(msg);
+
+export class PartyTime extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      room_number: '',
+      secret: '',
+      description: '',
+      show: false
+    }
+    this.setSecret = this.setSecret.bind(this);
+    this.setRoomNumber = this.setRoomNumber.bind(this);
+    this.setDescription = this.setDescription.bind(this);
+    this.newParty = this.newParty.bind(this);
+    this.handleClose = this.handleClose.bind(this);
+    this.handleOpen = this.handleOpen.bind(this);
+  }
+  handleClose() {
+    this.setState({show: false});
+  }
+  handleOpen() {
+    this.setState({show: true});
+  }
+  setSecret(event) {
+    this.setState({secret: event.target.value});
+  }
+  setRoomNumber(event) {
+    this.setState({room_number: event.target.value});
+  }
+  setDescription(event) {
+    this.setState({description: event.target.value});
+  }
+  newParty(event) {
+    event.preventDefault();
+    var data = {
+      room_number: this.state.room_number,
+      secret: this.state.secret,
+      description: this.state.description
+    }
+    axios.post(window.location.protocol + "//" + window.location.hostname + ":8000/api/party/", data)
+      .then(res => {
+	this.props.reload();
+	this.setState({show: false});
+      })
+      .catch((error) => {
+	if (error.response) {
+	  if (error.response.status == 400) {
+	    someError("Unable to create party: " + error.response.data);
```

Comment:
This error comes up if the name/email doesnt match
<img width="511" alt="image" src="https://github.com/Index01/RoomBot/assets/154588/42ba9d22-f0f0-4369-b24d-7f7584910588">
Do we want to guide people creating the party into finding the correct name/email for the room when they get it wrong? I'd say that this would be fine, with some possible collateral damage of randoms gaming the system

Reply:
Weird I wasn't able to repro this on staging - I got a `500` bc no `Guest` associated with that room. Spooky shit.

Did clean up the error handling though. Also throws a coherent error if the room already has a party.